astropy/units/tests/test_quantity.py

```diff
@@ -307,6 +307,12 @@ def test_creation_via_view(self):
         assert q6.unit == u.Unit(not_quite_a_foot)
         assert np.all(q6.to_value(u.cm) == 30. * a6)
 
+    def test_lshift_magnitude(self):
```

Comment:
I don't understand this comment: is that a test of converting a magnitude created with `<<` or something else?

Reply:
I meant something like
```
mag_fnu_cgs = u.mag(u.erg/u.s/u.cm**2/u.Hz)
m = 10. * u.mag(u.Jy)
m2 = m << mag_fnu_cgs
assert np.all(m2 == m.to(mag_fnu_cgs)
m <<= mag_fnu_cgs
assert np.all(m.value == m2.value)
assert m.unit == mag_fnu_cgs
```

Reply:
That code fails at the `<<=` line with this, even if I have a dummy `def __ilshift__()` in `FunctionUnitBase`:

```
        try:
>           factor = self.unit._to(other)
E           AttributeError: 'MagUnit' object has no attribute '_to'

astropy/units/quantity.py:882: AttributeError
```

Reply:
I think this is because `mag_fnu_cgs` is a `Unit`, so it's using `Quantity.__lshift__` instead of `FunctionUnit.__lshift__`:

```(Pdb) mag_fnu_cgs
Unit("mag(erg / (cm2 Hz s))")
```

Reply:
No, it should be a `MagUnit` (try `type(mag_fnu_cgs)`), and the fact that the first test works shows your code is OK.

But that the in-place one fails means that to avoid confusion, we should also implement `LogQuantity.__ilshift__`. I'm not sure whether my suggestion above would work, but we should be able to get at least the tests to pass with a simple:
```
def __ilshift__(self, other):
    # TODO: implement actual in-place!!
    return self.__lshift__(other)
```